kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -417,15 +421,24 @@ verifyClaimSentence sentence =
               | rejectRulePattern rulePattern -> True
             _ -> False
     rejectRulePattern
-        RulePattern { left, antiLeft, requires, rhs = RHS { right, ensures } }
+        RulePattern
+            { left
+            , antiLeft
+            , requires
+            , rhs = RHS { existentials, right, ensures }
+            }
       =
         let lhs = catMaybes [antiLeft] <> [left, unwrapPredicate requires]
             rhs = [right, unwrapPredicate ensures]
             freeVariablesUnion terms
                 = getFreeVariables
                     (foldMap freeVariables terms :: FreeVariables Variable)
+            existentialsInRHS = Set.fromList $ fmap ElemVar existentials
         in
-            not $ isSubsetOf (freeVariablesUnion rhs) (freeVariablesUnion lhs)
+            not 
+                $ isSubsetOf
+                    (freeVariablesUnion rhs \\ existentialsInRHS)
```

Comment:
Please add an instance of `HasFreeVariables` for `RHS` instead, so that `freeVariables` actually works correctly.